gramps/gen/utils/alive.py

```diff
@@ -89,234 +96,421 @@ def __init__(
+                    if ev_ref:
+                        evnt = self.db.get_event_from_handle(ev_ref.ref)
+                        if evnt and evnt.type.is_death_fallback():
+                            death_date_fb = evnt.get_date_object()
+                            death_found = True
+                            if death_date_fb.is_valid():
+                                death_date = death_date_fb
+                                explain_death = _("date fallback")
+                                if death_date.get_modifier() == Date.MOD_NONE:
+                                    death_date.set_modifier(Date.MOD_BEFORE)
+                                break  # we found a valid date, stop looking.
+            # At this point:
+            # * death_found is False: (no death indication found); or
+            # * death_found is True. (death confirmed somehow);  In which case:
+            #       * (death_date is valid) some form of death date found; or
+            #       * (death_date is None and no date was recorded)
+            # now repeat, looking for birth date
+            birth_ref = thisperson.get_birth_ref()
+            if birth_ref and birth_ref.get_role().is_primary():
+                evnt = self.db.get_event_from_handle(birth_ref.ref)
+                if evnt:
+                    dateobj = evnt.get_date_object()
+                    if dateobj and dateobj.is_valid():
+                        birth_date = dateobj
+                        explain_birth = _("date")
+
+            # to here:
+            #   birth_date is None: either no birth record or else no date reported; or
+            #   birth_date is a valid date
+            # Look for Baptism, etc events.
+            # These are fairly good indications of someone's birth date.
+            if not birth_date:
+                for ev_ref in thisperson.get_primary_event_ref_list():
+                    evnt = self.db.get_event_from_handle(ev_ref.ref)
+                    if evnt and evnt.type.is_birth_fallback():
+                        birth_date_fb = evnt.get_date_object()
+                        if birth_date_fb and birth_date_fb.is_valid():
+                            birth_date = birth_date_fb
+                            explain_birth = _("date fallback")
+                            break
+            if DEBUGLEVEL > 3:
+                LOG.debug(
+                    "           << get_person_bd for [%s], birth %s, death %s",
+                    thisperson.get_gramps_id(),
+                    birth_date,
+                    death_date,
+                )
+            return (birth_date, death_date, death_found, explain_birth, explain_death)
+
+        birth_date, death_date, known_to_be_dead, explain_birth_min, explain_death = (
+            get_person_bd(person)
+        )
+
+        explanation = (
+            _("DIRECT birth: ") + explain_birth_min + _(", death: ") + explain_death
```

Comment:
This is marked as translatable, but then DIRECT is used in a string comparison operation, which means that it would fail if translated.